pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -474,7 +542,7 @@ func (p *Pod) ConstructComposableWorkload(ctx context.Context, c client.Client,
 			return nil, err
 		}
 
-		if len(p.list.Items) != groupTotalCount {
+		if len(filterFailedPods(p.list.Items)) != groupTotalCount {
```

Comment:
This seems to be the only use of `filterFailedPods`. But we don't need the slice itself, we just need the count. It's better if we avoid allocating a slice just to count.